integration/utils/follower.go

```diff
@@ -82,11 +78,11 @@ func NewTxFollower(ctx context.Context, client *client.Client, opts ...followerO
 		ctx:    newCtx,
 		cancel: cancel,
 		logger: zerolog.Nop(),
-		mu:     &sync.RWMutex{},
 
-		inprogress: atomic.NewInt64(0),
+		mu:       &sync.RWMutex{},
+		txToChan: make(map[flowsdk.Identifier]txInfo),
 
-		stopped:  make(chan struct{}),
+		stopped:  make(chan struct{}, 1),
```

Comment:
no need?  you never insert into the channel

Reply:
o, could you remove `if f.height == 0 {` check below?  you've just initialized the follower, the height is always zero

Reply:
```
	for _, opt := range opts {
		opt(f)
	}
```

modifies f inline

Reply:
ah.  my bad